File: LeetCode/Java/26-two-pointers-删除排序数组中的重复项.java
/*
 * @lc app=leetcode.cn id=26 lang=java
 *
 * [26] 删除排序数组中的重复项
 */

// @lc code=start
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0)
            return 0;
        int j = 0;
        for (int i : nums)
            if (nums[j] != i)
                nums[++j] = i;
        return j + 1;
    }
}
// @lc code=end

File: LeetCode/C/26-two-pointers-删除排序数组中的重复项.cpp
/*
 * @lc app=leetcode.cn id=26 lang=cpp
 *
 * [26] 删除排序数组中的重复项
 */
#include "main.h"

// @lc code=start
class Solution {
  public:
    int removeDuplicates(vector<int> &nums) {
        if (nums.size() == 0)
            return 0;
        int p = 0;
        for (int num : nums)
            if (nums[p] != num)
                nums[++p] = num;
        return p + 1;
    }
};
// @lc code=end
